### PR TITLE
[WIP] change test_c10d_error_logger to use MultiThreadedTestCase

### PR DESCRIPTION
**Pull request by wz337**
Update test_c10d_error_logger.py to use MultiThreadedTestCase.

### CODE REVIEW

File: test/distributed/test_c10d_error_logger.py [hunk 1]
```diff
@@ -107,35 +47,31 @@ def failed_broadcast_not_raise_exception(self):
         except Exception as exception:
             pass
 
-    @with_comms
     def test_exception_handler_with_dist(self) -> None:
+        print(dist.get_rank())
```
Comment (wanchaol) [inline test/distributed/test_c10d_error_logger.py:51]: remove print?

File: test/distributed/test_c10d_error_logger.py [hunk 2]
```diff
@@ -107,35 +47,31 @@ def failed_broadcast_not_raise_exception(self):
...
+            dist._c10d_error_logger, level="DEBUG"
+        ) as captured:
             self.failed_broadcast_not_raise_exception()
             error_msg_dict = json.loads(
-                re.search("({.+})", captured.output[0]).group(0).replace("'", '"')
+                re.search("({.+})", captured.output[0])
+                .group(0)
+                .replace("'", '"')
             )
+            print(error_msg_dict)
```
Comment (wanchaol) [inline test/distributed/test_c10d_error_logger.py:64]: ditto

File: test/distributed/test_c10d_error_logger.py [hunk 3]
```diff
@@ -107,35 +47,31 @@ def failed_broadcast_not_raise_exception(self):
...
             self.assertEqual("broadcast", error_msg_dict["func_name"])
-
-            self.assertIn("args", error_msg_dict.keys())
-
-            self.assertIn("backend", error_msg_dict.keys())
-            self.assertEqual("nccl", error_msg_dict["backend"])
-
-            self.assertIn("world_size", error_msg_dict.keys())
+            self.assertEqual(f"(tensor([0, 1]), {self.world_size+1}), {{}}", error_msg_dict["args"])
+            self.assertEqual("threaded", error_msg_dict["backend"])
```
Comment (wanchaol) [inline test/distributed/test_c10d_error_logger.py:68]: I thought backend name for multithreaded pg is "local"?

File: test/distributed/test_c10d_error_logger.py [hunk 4]
```diff
@@ -107,35 +47,31 @@ def failed_broadcast_not_raise_exception(self):
...
-            self.assertIn("func_name", error_msg_dict.keys())
             self.assertEqual("broadcast", error_msg_dict["func_name"])
-
-            self.assertIn("args", error_msg_dict.keys())
-
-            self.assertIn("backend", error_msg_dict.keys())
-            self.assertEqual("nccl", error_msg_dict["backend"])
-
-            self.assertIn("world_size", error_msg_dict.keys())
+            self.assertEqual(f"(tensor([0, 1]), {self.world_size+1}), {{}}", error_msg_dict["args"])
```
Comment (wanchaol) [inline test/distributed/test_c10d_error_logger.py:67]: what's this new assert for?

**Review by wanchaol — COMMENTED** [review]
Is this currently working or suffering from deadlock?

File: test/distributed/test_c10d_error_logger.py (hunk 1 above)
Comment (wz337) [reply test/distributed/test_c10d_error_logger.py:51]: Ye. Thanks! Will clean up the prints after figuring out the hanging issue.

File: test/distributed/test_c10d_error_logger.py (hunk 4 above)
Comment (wz337) [reply test/distributed/test_c10d_error_logger.py:67]: For the exception handler, the logger actually logs the args and kwargs to the collective function. This is the args passed in from the broadcast call above. 

This assert was missing so adding it back here.

File: test/distributed/test_c10d_error_logger.py (hunk 3 above)
Comment (wz337) [reply test/distributed/test_c10d_error_logger.py:68]: Oh. Did it get updated? I am printing the name of backend locally and it is called "threaded".

**Pull request closed without merging**